sql/colexec: BenchmarkAggregator failed

[(sql/colexec).BenchmarkAggregator failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1713287&tab=buildLog) on [master@1a05ba6475f1b8d1f1de0c609a2a77e420c4c429](https://github.com/cockroachdb/cockroach/commits/1a05ba6475f1b8d1f1de0c609a2a77e420c4c429):

Fatal error:
```
panic: interface conversion: interface is nil, not colexec.resetter

```

Stack:
```
goroutine 813 [running]:
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*mergeJoinBase).completeBufferedGroup(0xc00042dc38, 0x26aa300, 0xc000052120, 0xc00042dc60, 0x26de240, 0xc000368b40, 0x0, 0x26a0014, 0xc000052120, 0x26de240, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/mergejoiner.go:508 +0xbf
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*mergeJoinBase).finishProbe(0xc00042dc38, 0x26aa300, 0xc000052120)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/mergejoiner.go:569 +0x70
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*mergeJoinInnerOp).Next(0xc00042dc38, 0x26aa300, 0xc000052120, 0x20, 0x26de240)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/mergejoiner_inner.eg.go:26042 +0x243
github.com/cockroachdb/cockroach/pkg/sql/colexec.BenchmarkMergeJoiner.func3(0xc000304fc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/mergejoiner_test.go:2197 +0x4ea
testing.(*B).runN(0xc000304fc0, 0x1)
	/usr/local/go/src/testing/benchmark.go:190 +0xcc
testing.(*B).run1.func1(0xc000304fc0)
	/usr/local/go/src/testing/benchmark.go:230 +0x64
created by testing.(*B).run1
	/usr/local/go/src/testing/benchmark.go:223 +0x7d

```

<details><summary>Log preceding fatal error</summary><p>

```
BenchmarkExternalSort/rows=320/cols=1/spilled=true-8                                                	       1	    206726 ns/op	  12.38 MB/s
BenchmarkExternalSort/rows=320/cols=2/spilled=false-8                                               	       1	     86122 ns/op	  59.45 MB/s
BenchmarkExternalSort/rows=320/cols=2/spilled=true-8                                                	       1	    280183 ns/op	  18.27 MB/s
BenchmarkExternalSort/rows=320/cols=4/spilled=false-8                                               	       1	    108344 ns/op	  94.51 MB/s
BenchmarkExternalSort/rows=320/cols=4/spilled=true-8                                                	       1	    338909 ns/op	  30.21 MB/s
BenchmarkExternalSort/rows=5120/cols=1/spilled=false-8                                              	       1	    247772 ns/op	 165.31 MB/s
BenchmarkExternalSort/rows=5120/cols=1/spilled=true-8                                               	       1	   1050531 ns/op	  38.99 MB/s
BenchmarkExternalSort/rows=5120/cols=2/spilled=false-8                                              	       1	    599713 ns/op	 136.60 MB/s
BenchmarkExternalSort/rows=5120/cols=2/spilled=true-8                                               	       1	   1901229 ns/op	  43.09 MB/s
BenchmarkExternalSort/rows=5120/cols=4/spilled=false-8                                              	       1	    856976 ns/op	 191.18 MB/s
BenchmarkExternalSort/rows=5120/cols=4/spilled=true-8                                               	       1	   2694946 ns/op	  60.80 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=false/distinct=true/rows=40-8                             	       1	    125158 ns/op	  20.45 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=false/distinct=true/rows=5120-8                           	       1	    640045 ns/op	 511.96 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=false/distinct=true/rows=81920-8                          	       1	  14025812 ns/op	 373.80 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=false/distinct=false/rows=40-8                            	       1	    110642 ns/op	  23.14 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=false/distinct=false/rows=5120-8                          	       1	    647994 ns/op	 505.68 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=false/distinct=false/rows=81920-8                         	       1	  13145196 ns/op	 398.84 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=true/distinct=true/rows=40-8                              	       1	    106692 ns/op	  23.99 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=true/distinct=true/rows=5120-8                            	       1	    721686 ns/op	 454.05 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=true/distinct=true/rows=81920-8                           	       1	  11960634 ns/op	 438.34 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=true/distinct=false/rows=40-8                             	       1	    102748 ns/op	  24.92 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=true/distinct=false/rows=5120-8                           	       1	    681563 ns/op	 480.78 MB/s
BenchmarkHashJoiner/nulls=false/fullOuter=true/distinct=false/rows=81920-8                          	       1	  12662851 ns/op	 414.04 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=false/distinct=true/rows=40-8                              	       1	     97191 ns/op	  26.34 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=false/distinct=true/rows=5120-8                            	       1	    907283 ns/op	 361.17 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=false/distinct=true/rows=81920-8                           	       1	  14602084 ns/op	 359.05 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=false/distinct=false/rows=40-8                             	       1	    112627 ns/op	  22.73 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=false/distinct=false/rows=5120-8                           	       1	    893657 ns/op	 366.67 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=false/distinct=false/rows=81920-8                          	       1	  15031149 ns/op	 348.80 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=true/distinct=true/rows=40-8                               	       1	    116471 ns/op	  21.98 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=true/distinct=true/rows=5120-8                             	       1	   1025108 ns/op	 319.65 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=true/distinct=true/rows=81920-8                            	       1	  15284330 ns/op	 343.02 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=true/distinct=false/rows=40-8                              	       1	    118683 ns/op	  21.57 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=true/distinct=false/rows=5120-8                            	       1	    907266 ns/op	 361.17 MB/s
BenchmarkHashJoiner/nulls=true/fullOuter=true/distinct=false/rows=81920-8                           	       1	  15688741 ns/op	 334.18 MB/s
BenchmarkLikeOps/selPrefixBytesBytesConstOp-8                                                       	       1	      3694 ns/op	 346.51 MB/s
BenchmarkLikeOps/selSuffixBytesBytesConstOp-8                                                       	       1	      3765 ns/op	 339.97 MB/s
BenchmarkLikeOps/selRegexpBytesBytesConstOp-8                                                       	       1	     35773 ns/op	  35.78 MB/s
BenchmarkColumnarizeMaterialize-8                                                                   	       1	   6476993 ns/op	  24.70 MB/s
BenchmarkMergeJoiner/rows=20-8                                                                      	       1	     93241 ns/op	  13.73 MB/s
BenchmarkMergeJoiner/rows=80-8                                                                      	       1	     64935 ns/op	  78.85 MB/s
BenchmarkMergeJoiner/rows=320-8                                                                     	       1	    103033 ns/op	 198.77 MB/s
BenchmarkMergeJoiner/rows=20480-8                                                                   	       1	   3242417 ns/op	 404.24 MB/s
BenchmarkMergeJoiner/oneSideRepeat-rows=20-8                                                        	       1	     54462 ns/op	  23.50 MB/s
BenchmarkMergeJoiner/oneSideRepeat-rows=80-8                                                        	       1	     69627 ns/op	  73.53 MB/s
BenchmarkMergeJoiner/oneSideRepeat-rows=320-8                                                       	       1	    105315 ns/op	 194.46 MB/s
BenchmarkMergeJoiner/oneSideRepeat-rows=20480-8                                                     	       1	   3286608 ns/op	 398.81 MB/s
BenchmarkMergeJoiner/bothSidesRepeat-rows=20-8                                                      	       1	     74781 ns/op	  17.12 MB/s
BenchmarkMergeJoiner/bothSidesRepeat-rows=80-8                                                      	       1	     79251 ns/op	  64.60 MB/s
BenchmarkMergeJoiner/bothSidesRepeat-rows=320-8                                                     	       1	    260585 ns/op	  78.59 MB/s

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=BenchmarkAggregator PKG=./pkg/sql/colexec TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ABenchmarkAggregator.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/colexec/mergejoiner_test.go
+++ b/pkg/sql/colexec/mergejoiner_test.go
@@ -25,6 +25,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+	"github.com/stretchr/testify/require"
 )
 
 type mjTestCase struct {
@@ -2085,27 +2086,18 @@ func BenchmarkMergeJoiner(b *testing.B) {
 				leftSource := newFiniteBatchSource(newBatchOfIntRows(nCols, batch), nBatches)
 				rightSource := newFiniteBatchSource(newBatchOfIntRows(nCols, batch), nBatches)
 
-				s := mergeJoinInnerOp{
-					mergeJoinBase{
-						allocator: testAllocator,
-						left: mergeJoinInput{
-							eqCols:      []uint32{0},
-							outCols:     []uint32{0, 1},
-							sourceTypes: sourceTypes,
-							source:      leftSource,
-							directions:  []execinfrapb.Ordering_Column_Direction{execinfrapb.Ordering_Column_ASC},
-						},
-
-						right: mergeJoinInput{
-							eqCols:      []uint32{0},
-							outCols:     []uint32{2, 3},
-							sourceTypes: sourceTypes,
-							source:      rightSource,
-							directions:  []execinfrapb.Ordering_Column_Direction{execinfrapb.Ordering_Column_ASC},
-						},
-					},
-				}
-
+				base, err := newMergeJoinBase(
+					testAllocator,
+					leftSource, rightSource,
+					[]uint32{0, 1}, []uint32{2, 3},
+					sourceTypes, sourceTypes,
+					[]execinfrapb.Ordering_Column{{ColIdx: 0, Direction: execinfrapb.Ordering_Column_ASC}},
+					[]execinfrapb.Ordering_Column{{ColIdx: 0, Direction: execinfrapb.Ordering_Column_ASC}},
+					nil,   /* filterConstructor */
+					false, /* filterOnlyOnLeft */
+				)
+				require.NoError(b, err)
+				s := mergeJoinInnerOp{mergeJoinBase: base}
 				s.Init()
 
 				b.StartTimer()
@@ -2127,26 +2119,18 @@ func BenchmarkMergeJoiner(b *testing.B) {
 				leftSource := newFiniteBatchSource(newBatchOfRepeatedIntRows(nCols, batch, nBatches), nBatches)
 				rightSource := newFiniteBatchSource(newBatchOfIntRows(nCols, batch), nBatches)
 
-				s := mergeJoinInnerOp{
-					mergeJoinBase{
-						allocator: testAllocator,
-						left: mergeJoinInput{
-							eqCols:      []uint32{0},
-							outCols:     []uint32{0, 1},
-							sourceTypes: sourceTypes,
-							source:      leftSource,
-							directions:  []execinfrapb.Ordering_Column_Direction{execinfrapb.Ordering_Column_ASC},
-						},
-
-						right: mergeJoinInput{
-							eqCols:      []uint32{0},
-							outCols:     []uint32{2, 3},
-							sourceTypes: sourceTypes,
-							source:      rightSource,
-							directions:  []execinfrapb.Ordering_Column_Direction{execinfrapb.Ordering_Column_ASC},
-						},
-					},
-				}
+				base, err := newMergeJoinBase(
+					testAllocator,
+					leftSource, rightSource,
+					[]uint32{0, 1}, []uint32{2, 3},
+					sourceTypes, sourceTypes,
+					[]execinfrapb.Ordering_Column{{ColIdx: 0, Direction: execinfrapb.Ordering_Column_ASC}},
+					[]execinfrapb.Ordering_Column{{ColIdx: 0, Direction: execinfrapb.Ordering_Column_ASC}},
+					nil,   /* filterConstructor */
+					false, /* filterOnlyOnLeft */
+				)
+				require.NoError(b, err)
+				s := mergeJoinInnerOp{mergeJoinBase: base}
 				s.Init()
 
 				b.StartTimer()
@@ -2170,27 +2154,18 @@ func BenchmarkMergeJoiner(b *testing.B) {
 				leftSource := newFiniteBatchSource(newBatchOfRepeatedIntRows(nCols, batch, numRepeats), nBatches)
 				rightSource := newFiniteBatchSource(newBatchOfRepeatedIntRows(nCols, batch, numRepeats), nBatches)
 
-				s := mergeJoinInnerOp{
-					mergeJoinBase{
-						allocator: testAllocator,
-						left: mergeJoinInput{
-							eqCols:      []uint32{0},
-							outCols:     []uint32{0, 1},
-							sourceTypes: sourceTypes,
-							source:      leftSource,
-							directions:  []execinfrapb.Ordering_Column_Direction{execinfrapb.Ordering_Column_ASC},
-						},
-
-						right: mergeJoinInput{
-							eqCols:      []uint32{0},
-							outCols:     []uint32{2, 3},
-							sourceTypes: sourceTypes,
-							source:      rightSource,
-							directions:  []execinfrapb.Ordering_Column_Direction{execinfrapb.Ordering_Column_ASC},
-						},
-					},
-				}
-
+				base, err := newMergeJoinBase(
+					testAllocator,
+					leftSource, rightSource,
+					[]uint32{0, 1}, []uint32{2, 3},
+					sourceTypes, sourceTypes,
+					[]execinfrapb.Ordering_Column{{ColIdx: 0, Direction: execinfrapb.Ordering_Column_ASC}},
+					[]execinfrapb.Ordering_Column{{ColIdx: 0, Direction: execinfrapb.Ordering_Column_ASC}},
+					nil,   /* filterConstructor */
+					false, /* filterOnlyOnLeft */
+				)
+				require.NoError(b, err)
+				s := mergeJoinInnerOp{mergeJoinBase: base}
 				s.Init()
 
 				b.StartTimer()

